docs/_static/js/search.js

```diff
@@ -0,0 +1,191 @@
+
+  const pageLink = document.createElement('a');
+  hitListElement.appendChild(pageLink);
+  pageLink.innerHTML = firstHierarchyLevel.value;
+  pageLink.href = pageURL;
+
+  const contextElement = document.createElement('div');
+  hitListElement.appendChild(contextElement);
+  contextElement.className = 'context';
+
+  if (lastHierarchyLevel && lastHierarchyLevel !== firstHierarchyLevel) {
+    const contextLinkContainer = document.createElement('div');
+    contextElement.appendChild(contextLinkContainer);
+    const contextLink = document.createElement('a');
+    contextLinkContainer.appendChild(contextLink);
+    contextLink.innerHTML = lastHierarchyLevel.value;
+    contextLink.href = anchorURL;
+  }
+
+  if (result.content) {
+    const contentElement = document.createElement('div');
+    contentElement.innerHTML = result.content.value;
+    contextElement.appendChild(contentElement);
+  }
+
+  return hitListElement;
+}
+
+function addResultsList(hits, query, parentElement) {
+  const searchResultsList = document.createElement('ul');
+  searchResultsList.className = 'search';
+  let hit;
+  for (hit of hits) {
+    const hitElement = createHitElement(hit, query);
+    searchResultsList.appendChild(hitElement);
+  }
+  parentElement.appendChild(searchResultsList);
+}
+
+function displayPagination(page, totalPages) {
+  const pagination = document.querySelector('#pagination');
+  const paginationItem = pagination.querySelector('.pagination-item');
+  const paginationPrevious = pagination.querySelector('.pagination-previous');
+  const paginationNext = pagination.querySelector('.pagination-next');
+
+  // Hide previous/next button if showing first/last page
+  pagination.querySelector('.pagination-previous').hidden = false;
+  pagination.querySelector('.pagination-previous').hidden = page === 0;
+  pagination.querySelector('.pagination-next').hidden = page === totalPages - 1;
+
+  // Display at most "toBeDisplayed" page links in the paginator
+  const toBeDisplayed = 7;
+  const [start, end] = setStartEndForPaginator(page, totalPages, toBeDisplayed);
+
+  for (let i = start; i < end; i += 1) {
+    const newPaginationbutton = document.createElement('button');
+    const previousPaginationbutton = document.createElement('button');
+    const nextPaginationbutton = document.createElement('button');
+    newPaginationbutton.classList.add('pagination-button');
+    previousPaginationbutton.classList.add('pagination-button');
+    nextPaginationbutton.classList.add('pagination-button');
+
+    let flag = false;
+    if (i === page) {
+      newPaginationbutton.innerHTML = 'Page ' + (i + 1) + ' of ' + end;
+      previousPaginationbutton.innerHTML = '← Previous';
+      nextPaginationbutton.innerHTML = 'Next →';
+      newPaginationbutton.setAttribute('aria-label', `page ${i + 1}`);
+      newPaginationbutton.setAttribute('aria-current', `${i}`);
+      flag = true;
+    }
+
+    // Register event handlers - Replaced onclick with AddEventListeners
```

Comment:
no need to add this extra bit to the comment here 

```suggestion
    // Register event listeners
```